lib/checkunusedvar.cpp

```diff
@@ -710,8 +710,7 @@ void CheckUnusedVar::checkFunctionVariableUsage_iterateScopes(const Scope* const
                      i->typeEndToken()->isStandardType() ||
                      isRecordTypeWithoutSideEffects(i->type()) ||
                      mSettings->library.detectContainer(i->typeStartToken(), /*iterator*/ false) ||
-                     (i->isStlType() &&
-                      !Token::Match(i->typeStartToken()->tokAt(2), "lock_guard|unique_lock|shared_ptr|unique_ptr|auto_ptr|shared_lock")))
+                     i->isStlType())
```

Comment:
Will this skip all `std` types? We should still warn for unused `std::string` or `std::vector`.

Reply:
This is an 'add' condition, not a 'skip'. Test case for unused string is here: https://github.com/danmar/cppcheck/blob/main/test/testunusedvar.cpp#L5255